node/pkg/reobserver/reobserver_test.go

```diff
@@ -0,0 +1,231 @@
+package reobserver
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	gossipv1 "github.com/certusone/wormhole/node/pkg/proto/gossip/v1"
+	"github.com/certusone/wormhole/node/pkg/vaa"
+
+	"github.com/ethereum/go-ethereum/common"
+	"go.uber.org/zap"
+)
+
+func TestMsgBeforeQuorum(t *testing.T) {
+	logger := zap.NewNop()
+	obsvReqSendC := make(chan *gossipv1.ObservationRequest, 50)
+	reob := NewReobserver(logger, obsvReqSendC)
+	assert.NotNil(t, reob)
+
+	msgId := "1/c69a1b1a65dd336bf1df6a77afb501fc25db7fc0938cb08595a9ef473265cb4f/1"
+
+	now := time.Now()
+	reob.AddMessage(msgId, vaa.ChainIDSolana, common.Hash{})
+	reob.QuorumReached(msgId)
+
+	// Make sure it was received and has been marked completed.
+	assert.Equal(t, 1, len(reob.observations))
+	oe, exists := reob.observations[msgId]
+	assert.Equal(t, true, exists)
+	assert.Equal(t, true, oe.localMsgReceived())
+	assert.Equal(t, true, oe.quorumReached)
+	assert.Equal(t, true, oe.completed)
+	assert.Equal(t, 0, oe.numRetries)
```

Comment:
For this sort of thing, would you mind just constructing an expected observationEntry struct and doing a single assertion against it?

Maybe something vaguely like this?

```
expected := observationEntry{
	msgId:		msgId
	chainId:	vaa.ChainIDSolana
	txHash:		common.Hash{}
	numRetries:	0
	quorumReached:	true
	completed:	true
}
assert.Equal(t, expected, oe)
```

I strongly preference this approach so if for some reason extra fields are added in the future, the test doesn't miss them. Worst case, just zero out the timestamp.